network_simulator/flow.py

```diff
@@ -89,5 +92,218 @@ def handle_packet_success(self, packet_id):
     def __repr__(self):
         return str(self.__dict__)
 
-# TODO(team): Flow-related Event subclasses
-# TODO(team): Flow subclasses for at least two TCP algorithms.
+
+
+class InitiateFlowEvent(Event):
+    def __init__(self, flow=None, packet_ids=[]):
+        """
+        An one-time Event called to set up the Flow at a given timestamp.
+
+        :ivar Flow flow: flow of this event.
+        :ivar list packet_ids: a list of integers of packet IDs.
+        :ivar list packets_to_send: list of data packets to send. This is
+        default to an empty list. This list will be populated later.
+        """
+        self.flow = flow
+        self.packet_ids = packet_ids
+        self.packets_to_send = []
+
+    def create_data_packet(self, packet_id, curr_time):
```

Comment:
Seems like this function is only used once so maybe just inline the DataPacket creation.